packages/react-components/react-drawer/src/components/Drawer/Drawer.types.ts

```diff
@@ -1,17 +1,64 @@
+import type { DialogProps, DialogSurfaceProps } from '@fluentui/react-dialog';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 
 export type DrawerSlots = {
+  /**
+   * Root slot of the Drawer.
+   * This slot is rendered as a `div` by default.
+   * Only renders if `type` is `inline`.
+   */
   root: Slot<'div'>;
 };
 
 /**
  * Drawer Props
  */
-export type DrawerProps = ComponentProps<DrawerSlots> & {};
+export type DrawerProps = ComponentProps<Partial<DrawerSlots>> & {
+  /**
+   * Position of the drawer.
+   * @defaultvalue 'left'
+   *
+   * - 'left' - Drawer is positioned on the left side of the screen.
+   * - 'right' - Drawer is positioned on the right side of the screen.
+   */
+  position?: 'left' | 'right';
+
+  /**
+   * Type of the drawer.
+   * @defaultvalue 'overlay'
+   *
+   * - 'overlay' - Drawer is hidden by default and can be opened by clicking on the trigger.
+   * - 'inline' - Drawer is stacked with the content
+   */
+  type?: 'inline' | 'overlay';
+
+  /**
+   * Size of the drawer.
+   * @defaultvalue 'small'
+   *
+   * - 'small' - Drawer is 320px wide.
+   * - 'medium' - Drawer is 592px wide.
+   * - 'large' - Drawer is 940px wide.
+   * - 'full' - Drawer is 100vw wide.
+   */
+  size?: 'small' | 'medium' | 'large' | 'full';
```

Comment:
There was a custom size in the spec. WIll it be, also, part of `size` prop here?

Reply:
As @Hotell and I discussed previously, this prop only enforces the preset sizes. In case any custom size is needed, a custom CSS can be created to override the default size. I've documented this in a [dedicated story](https://github.com/microsoft/fluentui/pull/27431/files#diff-ae0d98362a5d0b26eccaa13ac3e9bc3b0caa759c385b778067308252449322b8R32).